tests/tensor_search/integ_tests/test_custom_vector_field.py

```diff
@@ -0,0 +1,1099 @@
+                    type="float",
+                    features=[FieldFeature.ScoreModifier]),
+
+                # For searchable_attributes and filter tests
+                FieldRequest(
+                    name="my_custom_vector_2",
+                    type="custom_vector",
+                    features=[FieldFeature.Filter]),
+                FieldRequest(
+                    name="my_custom_vector_3",
+                    type="custom_vector",
+                    features=[FieldFeature.Filter]),
+
+                # For lexical + searchable_attributes test
+                FieldRequest(
+                    name="exact_field",
+                    type="text",
+                    features=[FieldFeature.LexicalSearch]),
+                FieldRequest(
+                    name="barely_field",
+                    type="text",
+                    features=[FieldFeature.LexicalSearch]),
+
+                # For multimodal mixed field tests
+                FieldRequest(
+                    name="multimodal_text",
+                    type="text"),
+                FieldRequest(
+                    name="multimodal_image",
+                    type="image_pointer"),
+                FieldRequest(
+                    name="my_multimodal",
+                    type="multimodal_combination",
+                    dependent_fields={"multimodal_text": 0.4, "multimodal_image": 0.6})
+            ],
+            tensor_fields=["my_custom_vector", "my_custom_vector_2", "my_custom_vector_3",
+                           "text_field", "my_multimodal"]
+        )
+
+        cls.indexes = cls.create_indexes([
+            unstructured_custom_index,
+            structured_custom_index
+        ])
+
+        cls.unstructured_custom_index = cls.indexes[0]
+        cls.structured_custom_index = cls.indexes[1]
+
+    def setUp(self):
+        super().setUp()
+        self.mappings = {
+            "my_custom_vector": {
+                "type": "custom_vector"
+            }
+        }
+
+        # Using arbitrary values so they're easy to eyeball
+        self.random_vector_1 = [1. for _ in range(512)]
+        self.random_vector_2 = [i*2 for i in range(512)]
+        self.random_vector_3 = [1 / (i + 1) for i in range(512)]
+
+        # Any tests that call add_document, search, bulk_search need this env var
+        self.device_patcher = mock.patch.dict(os.environ, {"MARQO_BEST_AVAILABLE_DEVICE": "cpu"})
+        self.device_patcher.start()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        self.device_patcher.stop()
+
+    def test_add_documents_with_custom_vector_field(self):
+        """
+        Add a document with a custom vector field:
+        mock call to vespa client
+        Result will be slightly different for unstructured vs structured indexes.
+        """
+
+        for index in self.indexes:
+            mock_feed_batch = mock.MagicMock()
+            mock_feed_batch.return_value = FeedBatchResponse(
```

Comment:
You're creating indexes anyway. Why mock and not do an integ test instead?

Reply:
I already have integ tests for these. These mocks are just extra insurance to make sure the internal calls are all correct. We used to do this a lot in v1.

Reply:
Happy to remove them, but I don't think these mock tests take up too much time though anyway, so it couldn't hurt to keep them.